src/dataService/savedItemsService.ts

```diff
@@ -422,6 +436,217 @@ export class SavedItemDataService {
+
+    await this.db('list')
+      .update({
+        status,
+        time_updated: SavedItemDataService.formatDate(new Date()),
+        api_id_updated: this.apiId,
+      })
+      .where({ item_id: itemId, user_id: this.userId });
+
+    return await this.getSavedItemById(itemId);
+  }
+
+  /**
+   * @param item
+   * @param savedItemUpsertInput
+   * @returns savedItem
+   */
+  public async upsertSavedItem(
+    item: ItemResponse,
+    savedItemUpsertInput: SavedItemUpsertInput
+  ): Promise<SavedItem> {
+    const currentDate = SavedItemDataService.formatDate(new Date());
+    const givenTimestamp = new Date(savedItemUpsertInput.timestamp * 1000);
+    const givenDate = savedItemUpsertInput.timestamp
+      ? SavedItemDataService.formatDate(givenTimestamp)
+      : currentDate;
+    //`returning` is not supported for mysql in knex
+    await this.db('list')
+      .insert({
+        user_id: parseInt(this.userId),
+        item_id: item.itemId,
+        given_url: savedItemUpsertInput.url,
+        status: 0,
+        resolved_id: item.resolvedId,
+        title: item.title,
+        time_added: givenDate,
+        time_updated: currentDate,
+        time_read: '0000-00-00 00:00:00',
+        time_favorited: savedItemUpsertInput.isFavorite
+          ? givenDate
+          : '0000-00-00 00:00:00',
+        favorite: savedItemUpsertInput.isFavorite ? 1 : 0,
+        api_id: parseInt(this.apiId),
+        api_id_updated: parseInt(this.apiId),
+      })
+      .onConflict()
+      .merge();
+
+    return await this.getSavedItemById(item.itemId.toString());
+  }
+
+  /**
+   * Build a query to update the `time_updated` field of many items
+   * the list table, by item id.
+   * @param itemIds The item IDS to update the `time_Updated` to now.
+   */
+  public updateListItemMany(itemIds: string[]): Knex.QueryBuilder {
+    return this.listItemUpdateBuilder().whereIn('item_id', itemIds);
+  }
+
+  /**
+   * Build a query to update the `time_updated` field of one item in
+   * the list table, by item id.
+   * @param itemId
+   */
+  public updateListItemOne(itemId: string): Knex.QueryBuilder {
+    return this.listItemUpdateBuilder().where('item_id', itemId);
+  }
+
+  /**
+   * Helper function to build updates to a user's list.
+   * Used to mark updates that affect the list item (e.g. a new tag
+   * association) but are not direct updates to the list table.
+   * Does not include the necessary `join` or `where` statement
+   * to properly execute this query.
+   * Do not run this query as-is. Should only be used to compose other
+   * queries. That's why it's private :)
+   */
+  private listItemUpdateBuilder(): Knex.QueryBuilder {
+    return this.db('list')
+      .update({
+        time_updated: SavedItemDataService.formatDate(new Date()),
+        api_id_updated: this.apiId,
+      })
+      .andWhere('user_id', this.userId);
+  }
+
+  public static createSavedItemDataServiceWithWriteDbClient(context: IContext) {
```

Comment:
please don't hate me. since it's in the same class now, could we update the name to `write` or `getWriteDbClient` or `writeDbClient`.
so `const savedItemService = SavedItemDataService.write()` for example.

100% also feel free to ignore this, I won't block on this nit

Reply:
done